Nos/Models/EditableNoteText.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  EditableNoteText.swift
+//  Nos
+//
+//  Created by Matthew Lorentz on 7/26/23.
+//
+
+import Foundation
+import UIKit
+
+/// A model for the unpublished `contents` field of a Nostr note. This model wraps an `AttributedString` and provides 
+/// functions to safely append raw `String` content as well as mentions of nostr entities like other notes or authors.
+struct EditableNoteText: Equatable {
+    
+    /// The underlying string with attributes like font and color.
+    private(set) var attributedString: AttributedString
+    
+    var nsAttributedString: NSAttributedString {
+        NSAttributedString(attributedString)
+    }
+    
+    var string: String {
+        nsAttributedString.string
+    }
+    
+    private static var font = UIFont.preferredFont(forTextStyle: .body)
+    
+    lazy var defaultAttributes: AttributeContainer = {
+        AttributeContainer(defaultNSAttributes)
+    }()
+    
+    var defaultNSAttributes: [NSAttributedString.Key: Any] = [
+        .font: font,
+        .foregroundColor: UIColor.primaryTxt
+    ]
+    
+    var isEmpty: Bool {
+        nsAttributedString.string.isEmpty
+    }
+    
+    // MARK: - Init
+    
+    init() {
+        self.attributedString = AttributedString()
+    }
+    
+    init(string: String) {
+        self.attributedString = AttributedString(string)
+    }
+    
+    init(nsAttributedString: NSAttributedString) {
+        self.attributedString = AttributedString(nsAttributedString)
+    }
+    
+    // MARK: - Modifying the contents
+    
+    /// Appends the given string and adds the the default styling attributes.
+    mutating func append(_ string: String) {
+        attributedString.append(AttributedString(string, attributes: defaultAttributes))
+    }
+    
+    /// Appends the given URL and adds the default styling attributes. Will append a space before the link if needed.
+    mutating func append(_ url: URL) {
+        if let lastCharacter = string.last,
+            !lastCharacter.isWhitespace {
+            append(" \(url.absoluteString)")
+        } else {
+            append(url.absoluteString)
```

Comment:
This could do something similar to insertMention and do instead:
`attributedString.append(AttributedString(url.absoluteString,attributes: defaultAttributes.merging(AttributeContainer([NSAttributedString.Key.link: url.absoluteString]))))`
I know it is long but it can be nicely formatted xd. That way the link will look like a link in the UI. The same for line 67.

Reply:
Yeah I could.

Reply:
I created an issue to track everything that remains to be done and to make sure that I address it.

- https://github.com/planetary-social/nos/issues/490